language/stdlib/modules/0L/Wallet.move

```diff
@@ -0,0 +1,411 @@
+  // The validator identifies the transaction by a unique id.
+  // tallies are computed on the fly, such that if a veto happens, the community which
+  // is faster than waiting for epoch boundaries.
+
+  public fun veto(sender: &signer, uid: u64) acquires CommunityTransfers, CommunityFreeze {
+    print(&0x110);
+    let addr = Signer::address_of(sender);
+    assert(
+      LibraSystem::is_validator(addr),
+      Errors::requires_role(ERR_PREFIX + 001)
+    );
+    print(&0x111);
+    let (opt, i) = find(uid, 0);
+    if (Option::is_some<TimedTransfer>(&opt)) {
+      let c = borrow_global_mut<CommunityTransfers>(0x0);
+      let t = Vector::borrow_mut<TimedTransfer>(&mut c.proposed, i);
+      Vector::push_back<address>(&mut t.veto.list, addr);
+      print(&0x112);
+
+      if (tally_veto(i)) {
+      print(&0x113);
+
+        reject(uid)
+      }
+    };
+  }
+
+  // private function. Once vetoed, the CommunityWallet transaction is remove from proposed list.
+  fun reject(uid: u64) acquires CommunityTransfers, CommunityFreeze {
+    print(&0x01131);
+    let c = borrow_global_mut<CommunityTransfers>(0x0);
+    let list = *&c.proposed;
+    let len = Vector::length(&list);
+    let i = 0;
+    print(&0x01132);
+    while (i < len) {
+      print(&0x01133);
+      let t = *Vector::borrow<TimedTransfer>(&list, i);
+      if (t.uid == uid) {
+        Vector::remove<TimedTransfer>(&mut c.proposed, i);
+        let f = borrow_global_mut<CommunityFreeze>(*&t.payer);
+        f.consecutive_rejections = f.consecutive_rejections + 1;
+        Vector::push_back(&mut c.rejected, t);
+        
+      };
+
+      i = i + 1;
+    };
+    
+    print(&0x01134);
+
+  }
+
+  // private function to tally vetos.
+  // checks if a voter is in the validator set.
+  // tallies everytime called. Only counts votes in the validator set.
+  // does not remove an address if not in the validator set, in case the validator returns
+  // to the set on the next tally.
+  fun tally_veto(index: u64): bool acquires CommunityTransfers {
+    let c = borrow_global_mut<CommunityTransfers>(0x0);
+    let t = Vector::borrow_mut<TimedTransfer>(&mut c.proposed, index);
+
+    let votes = 0;
+    let threshold = calculate_proportional_voting_threshold();
+    
+    let k = 0;
+    let len = Vector::length<address>(&t.veto.list);
+
+    while (k < len) {
+      let addr = *Vector::borrow<address>(&t.veto.list, k);
+      // ignore votes that are no longer in the validator set,
+      // BUT DON'T REMOVE, since they may rejoin the validator set, and shouldn't need to vote again.
+
+      if (LibraSystem::is_validator(addr)) {
+        votes = votes + NodeWeight::proof_of_weight(addr)
+      };
+      k = k + 1;
+    };
+
+    t.veto.count = votes;
+    t.veto.threshold = threshold;
+
+    return votes > threshold
+  }
+
+  // private function to get the total voting power of the validator set, and find the 2/3rds threshold
+  fun calculate_proportional_voting_threshold(): u64 {
```

Comment:
This is a very high bar to veto a transaction. Maybe that's fine, but under this scheme I doubt any transactions would get vetoed, we have enough trouble getting people to vote for upgrades.

Reply:
Agree, I think this parameter is a free variable. @zmanian  has some thoughts on this.